src/search-box/__tests__/search-box.spec.ts

```diff
@@ -43,4 +44,67 @@ describe('SearchBox', () => {
       expect(document.activeElement).not.toBe(widget.searchBox.nativeElement);
     });
   });
+
+  describe('[searchAsYouType]', () => {
+    describe('default behaviour', () => {
+      it('should call refine as your type by default', () => {
+        const fixture = createRenderer({
+          defaultState,
+          template: '<ais-search-box></ais-search-box>',
+          TestedWidget: NgAisSearchBox,
+        })();
+        const widget = fixture.componentInstance.testedWidget;
+        const refine = jest.spyOn(widget.state, 'refine');
+
+        const searchBox = widget.searchBox.nativeElement;
+        searchBox.value = 'the query';
+        searchBox.dispatchEvent(new Event('input'));
+        fixture.detectChanges();
+        expect(refine).toHaveBeenCalledTimes(1);
+        expect(refine).toHaveBeenCalledWith('the query');
+      });
+    });
+
+    describe('[searchAsYouType]="false"', () => {
+      let fixture;
+      let refine;
+
+      beforeEach(() => {
+        fixture = createRenderer({
+          defaultState,
+          template:
+            '<ais-search-box [searchAsYouType]="false"></ais-search-box>',
+          TestedWidget: NgAisSearchBox,
+        })();
+        const widget = fixture.componentInstance.testedWidget;
+        refine = jest.spyOn(widget.state, 'refine');
+
+        const searchBox = widget.searchBox.nativeElement;
+        searchBox.value = 'the query';
+        searchBox.dispatchEvent(new Event('input'));
+      });
+
+      it('should not call refine as you user is typing', () => {
+        expect(refine).toHaveBeenCalledTimes(0);
+      });
+
+      it('should call refine when submit button has been clicked', () => {
```

Comment:
This test is testing the browser behavior.

The following test sending the `submit` event already covers that one.

Reply:
Not totally true.
If for some reason we remove `type=submit` from this button, it will be broken.
You could say this test is there to make sure we haven't forgotten to add `type=submit` to the button.

Reply:
I understand your reasoning.

> You could say this test is there to make sure we haven't forgotten to add `type=submit` to the button.

We specifically target the button with `button[type=submit]` so this is not the purpose of this test.

If we remove `type=submit`, the search box won't be [spec](https://instantsearch-css.netlify.com/widgets/search-box/)-compliant anymore, so other tests should break before even testing the behavior.

Anyway, this is not a big deal!